src/components/Verification/ConnectionList.tsx

```diff
@@ -63,6 +73,154 @@ const ConnectionList = (props: {
+			/>
+		  </div>
+		);
+	  };
+
+const extractConnectionFields = (item: IConnectionList) => {
+	const connectionId = item?.connectionId || 'Not available';
+	const theirLabel = item?.theirLabel || 'Not available';
+	const createDateTime = item?.createDateTime || 'Not available';
+	return { connectionId, theirLabel, createDateTime };
+  };
+  
+  const isConnectionChecked = (connectionId: string) =>
+	localOrgs.map((item) => item.connectionId).includes(connectionId);
+
+  
+  const selectOrganization = async (item: IConnectionList, checked: boolean) => {
+	try {
+	  const { connectionId, theirLabel, createDateTime } = extractConnectionFields(item);
+	  const index = localOrgs?.findIndex((ele) => ele.connectionId === connectionId) ?? -1;
+  
+	  if (index === -1) {
+		setLocalOrgs((prev: LocalOrgs[]) => [
+		  ...prev,
+		  { connectionId, theirLabel, createDateTime },
+		]);
+	  } else if (!checked) {
+		const updateLocalOrgs = [...localOrgs];
+		updateLocalOrgs.splice(index, 1);
+		setLocalOrgs(updateLocalOrgs);
+	  }
+	} catch (error) {
+	  console.error('SELECTED ORGANIZATION:::', error);
+	}
+  };
+
+
+  const generateTable = async (connections: IConnectionList[]) => {
+	try {
+	  const connectionsData =
+		connections?.length > 0 &&
+		connections.map((ele: IConnectionList) => {
+		  const { connectionId, theirLabel, createDateTime } = extractConnectionFields(ele);
+		  const isChecked = isConnectionChecked(connectionId);
+  
+		  return {
+			data: [
+			  { data: renderCheckbox(ele, isChecked, connections) },
+			  { data: theirLabel },
+			  { data: connectionId },
+			  {
+				data: (
+				  <DateTooltip date={createDateTime} id="verification_connecetion_list">
+					{dateConversion(createDateTime)}
+				  </DateTooltip>
+				),
+			  },
+			],
+		  };
+		});
+  
+	  setTableData(connectionsData);
+	} catch (err) {
+	  console.error('Error generating table:', err);
+	}
+  };
+
+	useEffect(() => {
+		props.selectConnection(localOrgs);
+	}, [localOrgs]);
+
+
+	useEffect(() => {
+		generateTable(connectionList);
+	}, [connectionList, localOrgs]);
+
+	const updateLocalOrgs = async () => {
+		const res = await getFromLocalStorage(storageKeys.SELECTED_CONNECTIONS);
+		const selectedOrg = res ? JSON.parse(res) : [];
+		setLocalOrgs(selectedOrg);
+	};
+
+	useEffect(() => {
+		const clearStorageAndRefresh = async () => {
+			refreshPage();
+			await removeFromLocalStorage(storageKeys.SELECTED_CONNECTIONS);
+			await removeFromLocalStorage(storageKeys.SELECTED_USER);
+		};
+
+		clearStorageAndRefresh();
+	}, []);
+
+	useEffect(() => {
+		(async () => {
+			await setToLocalStorage(
+				storageKeys.SELECTED_CONNECTIONS,
+				JSON.stringify(localOrgs),
+			);
+		})();
+	}, [localOrgs]);
+
+
+	useEffect(() => {
+		let getData: NodeJS.Timeout;
+		updateLocalOrgs();
+		if (listAPIParameter?.search?.length >= 1) {
+			getData = setTimeout(() => {
```

Comment:
What is Data here? We can have relevant variable names.

Reply:
I will change name of variable accordingly